Runtime/Interaction/SnapZone.cs

```diff
@@ -317,16 +340,31 @@ public override void ProcessInteractor(XRInteractionUpdateOrder.UpdatePhase upda
             {
                 base.ProcessInteractor(updatePhase);
             }
-
+            
             if (socketActive)
             {
-                if (m_HoverTargets.Count == 0 && ShowHighlightObject)
+                hoverTargets.RemoveAll(target => target == null || target.enabled == false);
+                hoverTargets.First(target =>
+                {
+                    if (m_HoverTargets.Contains(target) == false && target.isSelected == false && selectTarget == null)
+                    {
+                        if (CanSelect(target))
+                        {
+                            ForceSelect(target);
+                            return true;
+                        }
+                    }
+
+                    return false;
+                });
+                
+                if (hoverTargets.Count == 0 && ShowHighlightObject)
                 {
                     activeMaterial = HighlightMeshMaterial;
                 } 
-                else if (m_HoverTargets.Count > 0 && showInteractableHoverMeshes)
+                else if (hoverTargets.Count > 0 && showInteractableHoverMeshes)
```

Comment:
What are `m_HoverTargets` and how they are replaced by `hoverTargets`?
My main concern is: do they consider the same elements in their list? I don't think they contain the same elements.

Reply:
hoverTargets does contain the same elements than hoverTargets, but also the additional ones which are lost due to badly implemented trigger implementation of unity xr

Reply:
So this is an improved `hoverTargets`? If so, I am fine with it. I just wanted to alert that we were not missing objects in our implementation.

Reply:
yes, has multiple entries of the same object (for each collider entered) which allows us to still keep track of the object when one of the collider triggers an exit.